modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -1213,31 +1215,47 @@ PolygonMeshGeneratorBase::buildSimplePeripheral(
   {
     for (unsigned int j = 0; j < num_sectors_per_side; j++)
     {
-      Elem * elem_Quad4 = mesh->add_elem(new Quad4);
-      elem_Quad4->set_node(0) = mesh->node_ptr(j + (num_sectors_per_side + 1) * i);
-      elem_Quad4->set_node(1) = mesh->node_ptr(j + 1 + (num_sectors_per_side + 1) * i);
-      elem_Quad4->set_node(2) = mesh->node_ptr(j + 1 + (num_sectors_per_side + 1) * (i + 1));
-      elem_Quad4->set_node(3) = mesh->node_ptr(j + (num_sectors_per_side + 1) * (i + 1));
-      elem_Quad4->subdomain_id() = PERIPHERAL_ID_SHIFT + id_shift;
+      std::unique_ptr<Elem> new_elem;
+
+      new_elem = std::make_unique<Quad4>();
+      new_elem->set_node(0) = mesh->node_ptr(j + (num_sectors_per_side + 1) * (i));
+      new_elem->set_node(1) = mesh->node_ptr(j + 1 + (num_sectors_per_side + 1) * (i));
+      new_elem->set_node(2) = mesh->node_ptr(j + 1 + (num_sectors_per_side + 1) * (i + 1));
+      new_elem->set_node(3) = mesh->node_ptr(j + (num_sectors_per_side + 1) * (i + 1));
+
+      Elem * elem = mesh->add_elem(std::move(new_elem));
+
+      // add subdoamin/boundary IDs
+      elem->subdomain_id() = PERIPHERAL_ID_SHIFT + id_shift;
       if (i == 0)
       {
-        boundary_info.add_side(elem_Quad4, 0, OUTER_SIDESET_ID);
+        boundary_info.add_side(elem, 0, OUTER_SIDESET_ID);
         if (create_inward_interface_boundaries)
-          boundary_info.add_side(elem_Quad4, 0, SLICE_ALT + id_shift * 2);
+          boundary_info.add_side(elem, 0, SLICE_ALT + id_shift * 2);
       }
       if (i == peripheral_invervals - 1)
       {
-        boundary_info.add_side(elem_Quad4, 2, OUTER_SIDESET_ID);
+        boundary_info.add_side(elem, 2, OUTER_SIDESET_ID);
         if (create_outward_interface_boundaries)
-          boundary_info.add_side(elem_Quad4, 2, SLICE_ALT + id_shift * 2 + 1);
+          boundary_info.add_side(elem, 2, SLICE_ALT + id_shift * 2 + 1);
       }
       if (j == 0)
-        boundary_info.add_side(elem_Quad4, 3, OUTER_SIDESET_ID);
+        boundary_info.add_side(elem, 3, OUTER_SIDESET_ID);
       if (j == num_sectors_per_side - 1)
-        boundary_info.add_side(elem_Quad4, 1, OUTER_SIDESET_ID);
+        boundary_info.add_side(elem, 1, OUTER_SIDESET_ID);
     }
   }
 
+  mesh->remove_orphaned_nodes();
+
+  // convert element to second order if needed
```

Comment:
is this converting only the boundary or boundary + pattern?

Reply:
This should convert only the boundary. This mesh should be stitched to the pattern after returning from this function.